### PR TITLE
Expose SecurityContext in FlinkClusterSpec

### PR DESCRIPTION
**Pull request by thebalu**
Expose the PodSecurityContext for JobManager, TaskManager, and Job pods in the FlinkClusterSpec.
This lets us run them as the Flink user, as mentioned in https://github.com/GoogleCloudPlatform/flink-on-k8s-operator/issues/312

### CODE REVIEW

**Comment by thebalu** [comment]
Hi! This is my first time contributing to this project, so please let me know if there are any additional requirements.

I am not sure for example, if this requires an API version change; if so, could you please provide some guidance on how to do it?

**Comment by thebalu** [comment]
@googlebot I signed it!

**Review by functicons — CHANGES_REQUESTED** [review]
Thanks for the PR! 

1. Could you update the [CRD doc](https://github.com/GoogleCloudPlatform/flink-on-k8s-operator/blob/master/docs/crd.md) with the new fields? 

2. Update the unit test to cover cases with and without the new fields.

3. I'd prefer keeping the sample YAMLs simple. This feature is not so frequently requested, so I don't want to confuse people with them in the sample YAMLs. For users needing it, they can go to the CRD doc.

**Comment by thebalu** [comment]
@functicons I've made the requested changes, please review when you have time.

File: docs/crd.md [hunk 1]
```diff
@@ -40,6 +40,7 @@ FlinkCluster
         |__ sidecars
         |__ podAnnotations
         |__ podLabels
+        |__ securityContext
```
Comment (functicons) [inline docs/crd.md:43]: Could you add explanation for these fields below?

**Review by functicons — CHANGES_REQUESTED** [review]
A few minor comments.

Could you also add a section in the [user guide](https://github.com/GoogleCloudPlatform/flink-on-k8s-operator/blob/master/docs/user_guide.md#how-to) on the use case and how to set these fields?

**Comment by thebalu** [comment]
@functicons updated docs

**Review by functicons — APPROVED** [review]
Thanks!

**Pull request merged**